diff -ru a/etc/init.d/rcS b/etc/init.d/rcS
--- a/etc/init.d/rcS	2022-11-04 10:02:34.000000000 +0100
+++ b/etc/init.d/rcS	2022-11-04 16:15:46.000000000 +0100
@@ -108,7 +108,12 @@
 FS_CORRUPT=0
 dosfsck -a -w $ONBOARD_PARTITION || dosfsck -a -w $ONBOARD_PARTITION || dosfsck -a -w $ONBOARD_PARTITION || dosfsck -a -w $ONBOARD_PARTITION || FS_CORRUPT=1
 [ ! -s /usr/local/Kobo/pickel ] && FORCE_FACTORY_RESET=1
-/usr/local/Kobo/pickel should-factory-reset
+PLATO_HOME=/mnt/onboard/.adds/plato
+if [ -e "$PLATO_HOME"/bootlock ]; then
+	true
+else
+	/usr/local/Kobo/pickel should-factory-reset
+fi
 if [ $? != 0 ] || [ $FS_CORRUPT == 1 ] || [ $FORCE_FACTORY_RESET == 1 ]; then
 	case $PRODUCT in
 		kraken|phoenix)
@@ -250,7 +255,12 @@
 /bin/dbus-daemon --system &
 export DBUS_SESSION_BUS_ADDRESS=`/bin/dbus-daemon --session --print-address --fork`
 
-/usr/local/Kobo/hindenburg &
-LIBC_FATAL_STDERR_=1 /usr/local/Kobo/nickel -platform kobo -skipFontLoad &
+if [ -e "$PLATO_HOME"/bootlock -a -x "$PLATO_HOME"/plato.sh ]; then
+	PLATO_STANDALONE=1 "$PLATO_HOME"/plato.sh &
+else
+	[ -e "$PLATO_HOME"/bootlock ] && rm "$PLATO_HOME"/bootlock
+	/usr/local/Kobo/hindenburg &
+	LIBC_FATAL_STDERR_=1 /usr/local/Kobo/nickel -platform kobo -skipFontLoad &
+fi
 [ $PLATFORM != freescale ] && rm -rf /dev/mmcblk1* && udevadm trigger &
 [ -e /drivers/$PLATFORM/misc/lowmem.ko ] && insmod /drivers/$PLATFORM/misc/lowmem.ko &
