diff -ru a/Makerules b/Makerules
--- a/Makerules	2026-01-09 12:19:57
+++ b/Makerules	2026-01-09 12:19:36
@@ -354,6 +354,35 @@
 # This does rely on the generated directory being populated with the font files.
 # Run 'make generate' before doing the cross compile.
 
+ifeq "$(OS)" "kobo"
+  CC = arm-linux-gnueabihf-gcc
+  CXX = arm-linux-gnueabihf-g++
+  LD = arm-linux-gnueabihf-ld
+  AR = arm-linux-gnueabihf-ar
+  AS = arm-linux-gnueabihf-as
+  CFLAGS += -O2 -fPIC -mcpu=cortex-a9 -mfpu=neon -DTOFU_CJK_LANG -DTOFU_CJK_EXT -DFZ_ENABLE_ICC=0 -DFZ_ENABLE_SPOT_RENDERING=0 -DFZ_ENABLE_ODT_OUTPUT=0 -DFZ_ENABLE_OCR_OUTPUT=0
+  HAVE_PTHREAD := yes
+  HAVE_LIBCRYPTO := no
+  HAVE_X11 := no
+  HAVE_GLFW := no
+  SYS_PTHREAD_CFLAGS :=
+  SYS_PTHREAD_LIBS := -lpthread
+  SYS_FREETYPE_CFLAGS := -I../freetype2/include
+  SYS_FREETYPE_LIBS := -L../freetype2/objs/.libs -lfreetype
+  SYS_GUMBO_CFLAGS := -I../gumbo/src
+  SYS_GUMBO_LIBS := -L../gumbo/.libs -lgumbo
+  SYS_HARFBUZZ_CFLAGS := -I../harfbuzz/src
+  SYS_HARFBUZZ_LIBS := -L../harfbuzz/build/src -lharfbuzz
+  SYS_OPENJPEG_CFLAGS := -I../openjpeg/src/lib/openjp2
+  SYS_OPENJPEG_LIBS := -L../openjpeg/build/bin -lopenjpeg
+  SYS_JBIG2DEC_CFLAGS := -I../jbig2dec
+  SYS_JBIG2DEC_LIBS := -L../jbig2dec/.libs -ljbig2dec
+  SYS_LIBJPEG_CFLAGS := -I../libjpeg
+  SYS_LIBJPEG_LIBS := -L../libjpeg/.libs -ljpeg
+  SYS_ZLIB_CFLAGS := -I../zlib
+  SYS_ZLIB_LIBS := -L../zlib -lz
+endif
+
 ifeq "$(OS)" "wasm"
   build_prefix += wasm/
   CC = emcc
diff -ru a/Makethird b/Makethird
--- a/Makethird	2026-01-09 12:19:57
+++ b/Makethird	2026-01-09 12:19:36
@@ -83,10 +83,6 @@
   endif
 endif
 
-# --- Include source file lists ---
-
-include Makelists
-
 # --- BROTLI ---
 
 ifeq ($(USE_BROTLI),yes)
